sdk/storage/azblob/zc_appendBlobClient.go

```diff
@@ -25,36 +25,71 @@ func NewAppendBlobClient(blobURL string, cred azcore.Credential, options *connec
 		return AppendBlobClient{}, err
 	}
 	con := newConnection(blobURL, cred, options)
-	return AppendBlobClient{client: &appendBlobClient{con: con}, u: *u}, nil
-}
-
-func (ab AppendBlobClient) WithPipeline(pipeline azcore.Pipeline) AppendBlobClient {
-	con := newConnectionWithPipeline(ab.u.String(), pipeline)
-	return AppendBlobClient{client: &appendBlobClient{con}, u: ab.u}
-}
-
-func (ab AppendBlobClient) GetAccountInfo(ctx context.Context) (BlobGetAccountInfoResponse, error) {
-	blobClient := BlobClient{client: &blobClient{ab.client.con, nil}}
-
-	return blobClient.GetAccountInfo(ctx)
+	return AppendBlobClient{
+		client:     &appendBlobClient{con: con},
+		u:          *u,
+		BlobClient: BlobClient{client: &blobClient{con: con}},
+	}, nil
 }
 
 func (ab AppendBlobClient) URL() url.URL {
 	return ab.u
 }
 
+func (ab AppendBlobClient) WithPipeline(pipeline azcore.Pipeline) AppendBlobClient {
+	con := newConnectionWithPipeline(ab.u.String(), pipeline)
+	return AppendBlobClient{
+		client:     &appendBlobClient{con},
+		u:          ab.u,
+		BlobClient: BlobClient{client: &blobClient{con: con}},
+	}
+}
+
+// WithSnapshot creates a new AppendBlobURL object identical to the source but with the specified snapshot timestamp.
+// Pass "" to remove the snapshot returning a URL to the base blob.
 func (ab AppendBlobClient) WithSnapshot(snapshot string) AppendBlobClient {
 	p := NewBlobURLParts(ab.URL())
 	p.Snapshot = snapshot
 	snapshotURL := p.URL()
+	con := newConnectionWithPipeline(snapshotURL.String(), ab.client.con.p)
+	return AppendBlobClient{
+		client:     &appendBlobClient{con: con},
+		u:          ab.u,
+		BlobClient: BlobClient{client: &blobClient{con: con}},
+	}
+}
+
+// WithVersionID creates a new AppendBlobURL object identical to the source but with the specified version id.
+// Pass "" to remove the versionID returning a URL to the base blob.
+func (ab AppendBlobClient) WithVersionID(versionID string) AppendBlobClient {
+	p := NewBlobURLParts(ab.URL())
+	p.VersionID = versionID
+	versionIDURL := p.URL()
+	con := newConnectionWithPipeline(versionIDURL.String(), ab.client.con.p)
 	return AppendBlobClient{
-		client: &appendBlobClient{
-			newConnectionWithPipeline(snapshotURL.String(), ab.client.con.p),
-		},
-		u: ab.u,
+		client:     &appendBlobClient{con: con},
+		u:          ab.u,
+		BlobClient: BlobClient{client: &blobClient{con: con}},
 	}
 }
 
+func (ab AppendBlobClient) GetAccountInfo(ctx context.Context) (BlobGetAccountInfoResponse, error) {
+	blobClient := BlobClient{client: &blobClient{ab.client.con, nil}}
```

Comment:
We shouldn't need to create a blobClient, refer to track 1.5 where we call the parent directly. But actually these methods don't have to be created on the children, right? They are all inherited.

Reply:
Yes that's correct. We can directly inherit the method from parent. I wonder why we have those methods in track 1.5 too